broker.go

```diff
@@ -118,6 +118,11 @@ func (b *Broker) Open(conf *Config) error {
 		}
 
 		if conf.Net.SASL.Enable {
+			b.connErr = b.sendAndReceiveSASLPlainHandshake()
+			if b.connErr != nil {
+				Logger.Printf("Error while performing SASL handshake %s\n", b.addr)
+				return
```

Comment:
Thanks for reviewing it @wvanbergen 

I might have miss something here. I guess that the connection should also be closed and the "b.opened" flag set to 0. 

However as we are in a goroutine, I don't understand why the err variable should be set. We just stop the connection routine and the b.connErr is used to detect and error.